larray/viewer.py

```diff
@@ -2252,22 +2281,135 @@ def set_widget_array(self, array, title):
+
+    def _askToSaveIfDataModified(self):
+        if self._isDataModified():
+            ret = QMessageBox.warning(self, "Warning", "The data have been modified.\nDo you want to save your changes?",
+                               QMessageBox.Save | QMessageBox.Discard | QMessageBox.Cancel)
+            if ret == QMessageBox.Save:
+                self.apply_changes()
+                return self.save()
+            elif ret == QMessageBox.Cancel:
+                return False
+            else:
+                return True
+        else:
+            return True
+
+    @Slot()
+    def new(self):
+        if self._askToSaveIfDataModified():
+            self._listwidget.clear()
+            self.data = la.Session()
+            self.arraywidget.set_data(la.zeros(0))
+            self.setCurrentFile(None)
+            self.statusBar().showMessage("Viewer has been reset", 2000)
+
+    def _openFile(self, filepath):
+        # XXX : clear console history ?
+        self._listwidget.clear()
+        self.data = la.Session(filepath)
+        arrays = [k for k, v in self.data.items() if self._display_in_grid(k, v)]
+        self.add_list_items(arrays)
+        self._listwidget.setCurrentRow(0)
+        self.setCurrentFile(filepath)
+        self.statusBar().showMessage("File {} loaded".format(os.path.basename(filepath)), 2000)
+
+    @Slot()
+    def open(self):
+        if self._askToSaveIfDataModified():
+            filepath = QFileDialog.getOpenFileName(self)
+            if filepath:
+                self._openFile(filepath)
+            else:
+                QMessageBox.warning(self, "Warning", "No file selected")
+
+    @Slot()
+    def openRecentFile(self):
+        if self._askToSaveIfDataModified():
+            action = self.sender()
+            if action:
+                filepath = action.data()
+                self._openFile(filepath)
+
+    def _saveData(self, filepath):
+        self.data.dump(filepath)
+        self.setCurrentFile(filepath)
+        self._appliedchanges = False
+        self.statusBar().showMessage("Arrays saved in file {}".format(filepath), 2000)
+
+    @Slot()
+    def save(self):
+        if self.currentFile is not None:
+            self._saveData(self.currentFile)
+        else:
+            self.saveAs()
+        return True
+
+    @Slot()
+    def saveAs(self):
+        dialog = QFileDialog(self)
+        dialog.setWindowModality(Qt.WindowModal)
+        dialog.setAcceptMode(QFileDialog.AcceptSave)
+        if dialog.exec() != QDialog.Accepted:
+            QMessageBox.critical(self, "Error", "Current session could not be saved")
+            return False
+        else:
+            self._saveData(dialog.selectedFiles()[0])
+            return True
+
+    @Slot()
+    def about(self):
+        QMessageBox.about(self, _("About"), _("Larray's viewer. "
+                        "See <a href=\"http://larray.readthedocs.io/en/stable/\">larray_documentation</a> "
```

Comment:
I am not against an "about" window per se, but this one would be better as a direct "online documentation" link in the "help" menu (which is a great thing to have btw).